pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -148,30 +149,34 @@ func (c *Cluster) configureDashboardModuleSettings(daemonID string) (bool, error
 	return hasChanged, nil
 }
 
-func (c *Cluster) initializeSecureDashboard() (bool, error) {
+func (c *Cluster) initializeSecureDashboard() error {
 	// we need to wait a short period after enabling the module before we can call the `ceph dashboard` commands.
 	time.Sleep(dashboardInitWaitTime)
 
 	password, err := c.getOrGenerateDashboardPassword()
 	if err != nil {
-		return false, errors.Wrap(err, "failed to generate a password for the ceph dashboard")
+		return errors.Wrap(err, "failed to generate a password for the ceph dashboard")
 	}
 
 	if c.spec.Dashboard.SSL {
 		alreadyCreated, err := c.createSelfSignedCert()
 		if err != nil {
-			return false, errors.Wrap(err, "failed to create a self signed cert for the ceph dashboard")
+			return errors.Wrap(err, "failed to create a self signed cert for the ceph dashboard")
 		}
 		if alreadyCreated {
-			return false, nil
+			return nil
+		} else {
```

Comment:
nit: no need for the `else` block since there was a `return` statement from the previous condition.
```suggestion
		} 
```